### PR TITLE
Bump OpenTelemetry packages to 1.8.0 releases

### LINKED ISSUE #3328: Upgrade OTel SDK to 1.8.0
OTel .NET SDK 1.8.0 is expected to be released in April.

We need to upgrade it.

Changes required adjustments on our side (it is probably not full list):
- https://github.com/open-telemetry/opentelemetry-dotnet/pull/5448 - native support for detecting samplers. We should be able to remove code from our side

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #3328

## What

Bump OpenTelemetry packages to 1.8.0 releases
Move support for `OTEL_TRACES_SAMPLER` and `OTEL_TRACES_SAMPLER_ARG` to SDK by https://github.com/open-telemetry/opentelemetry-dotnet/pull/5448

## Tests

CI

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] ~~New~~ features are covered by tests.

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -13,24 +13,24 @@
...
+    <PackageVersion Include="OpenTelemetry.Instrumentation.AspNet" Version="1.8.0-beta.1" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.8.0" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.EntityFrameworkCore" Version="1.0.0-beta.11" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.GrpcNetClient" Version="1.8.0-beta.1" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.Process" Version="0.5.0-beta.5" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.Quartz" Version="1.0.0-beta.2" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.Runtime" Version="1.8.0" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.SqlClient" Version="1.8.0-beta.1" />
+    <PackageVersion Include="OpenTelemetry.Instrumentation.StackExchangeRedis" Version="1.0.0-rc9.14" />
     <PackageVersion Include="OpenTelemetry.Shims.OpenTracing" Version="1.7.0-beta.1" />
```
Comment (Kielek) [inline src/Directory.Packages.props:28]: What is your opinion about this package?
There is no changes sinse 1.7.0-beta.1 release. The only thing we can achieve is upgrade transitive dependency from OpenTelemetry.Api 1.7.0 to 1.8.0.

We ship 1.8.0 even without this change.

Comment (rajkumar-rangaraj) [reply src/Directory.Packages.props:28]: I recommend shipping without any additional changes. These are not stable packages; we should rely on .NET to transitively upgrade the dependencies for these libraries from 1.7.0 to 1.8.0.

It’s a maintenance overhead for the SDK team to just modify these package dependencies to 1.8.0 and release another version.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by pjanotti — APPROVED** [review]
:shipit:

**Pull request merged**